src/timeout.ts

```diff
@@ -171,7 +171,7 @@ function isCSOTTimeoutContextOptions(v: unknown): v is CSOTTimeoutContextOptions
 
 /** @internal */
 export abstract class TimeoutContext {
-  static create(options: TimeoutContextOptions): TimeoutContext {
+  static create(options: Partial<TimeoutContextOptions>): TimeoutContext {
```

Comment:
No, I believe that was an oversight.  Why was it required in the first place?

Reply:
Isn't it always necessary to provide because serverSelection is the max bound of that step? @W-A-James may be more familiar than I

Reply:
Eventually I think we'll want to create timeout contexts that apply to subparts of the driver.  For example, authentication and monitoring execute commands directly on the connection.  I envisioned us creating timeout with `TimeoutContext.create()` and passing the result in directly.

These places do not have a server selection timeout provided and do not need one.

Reply:
Since this isn't how we use it now can we keep making sure we pass in this value to not make any mistakes for the current use cases? making it Partial later when there is no need for it shouldn't be too much effort, but we're still WIP on the CSOT use case that does need it.